docs/data/date-pickers/lifecycle/lifecycle.md

```diff
@@ -0,0 +1,346 @@
+---
+product: date-pickers
+title: Components lifecycle
+githubLabel: 'component: pickers'
+packageName: '@mui/x-date-pickers'
+---
+
+# Components lifecycle
+
+<p class="description">This page explains when the onChange, onAccept, and onClose callbacks are called.</p>
+
+## Fields lifecycle
+
+### On simple fields
+
+:::info
+The information below is applicable to standalone fields (when rendering `<DateField />`),
+as well as to pickers for field editing (when rendering `<DatePicker />` and using the input to edit the value).
+:::
+
+The field components have an internal state controlling the visible value.
+
+It will only call the `onChange` callback when:
+
+- the user fills one section of an empty field. The value equals `Invalid date`.
+- the user completes all sections of a field. The value reflects the field.
+- the user cleans one section of a completed field. The value equals `Invalid date`.
+- the user cleans all sections of a field. The value equals `null`.
+
+The example below shows the last value received by `onChange`.
+
+{{"demo": "LifeCycleDateFieldEmpty.js", "defaultCodeOpen": false}}
+
+### On range fields [<span class="plan-pro"></span>](/x/introduction/licensing/#pro-plan)
+
+On range fields (`SingleInputDateRangeField` / `MultiInputDateRangeField` / ... ),
+`onChange` is called if the date you are modifying is matching one of the conditions above,
+regardless of the other date state.
+
+The example below shows the last value received by `onChange`.
+Note how changing the value of the start date section will call `onChange` even if the end date is empty or partially filled.
+
+{{"demo": "LifeCycleDateRangeField.js", "defaultCodeOpen": false}}
+
+## Pickers lifecycle
+
+### When is `onClose` called?
+
+:::info
+In all the below scenarios, the picker closes when `onClose` is called, except if you are controlling the `open` prop.
+:::
+
+#### When the last view is completed
+
+When a selection in the last view is made, `onClose` will be called only if the `closeOnSelect` prop is equal to `true`.
+By default, it is set to `true` on desktop and `false` on mobile.
+
+Here are a few examples:
+
+:::info
+The examples below are using the desktop and mobile variants of the pickers, but the behavior is exactly the same when using the responsive variant (`DatePicker`, `TimePicker`, ...) on a mobile or desktop environment.
+:::
+
+- ```tsx
+  <DesktopDatePicker />
+  ```
+
+  - Default `views` prop: `['year', 'day']`
+  - Default `closeOnSelect` prop: `true`
+
+  **Behavior:** The picker will close when selecting the day.
+
+- ```tsx
+  <DesktopDatePicker closeOnSelect={false} />
+  ```
+
+  - Default `views` prop: `['year', 'day']`
+  - Explicit `closeOnSelect` prop: `false`
+
+  **Behavior:** The picker won't close when selecting the day. The user will have to click on the _OK_ action to close it.
```

Comment:
Change only to avoid an extra line. 🙈 
```suggestion
  **Behavior:** The picker won't close when selecting a day. The user will have to click on the _OK_ action to close it.
```